### ISSUE TITLE
[5.0.22] passenger-install-apache2 fails on Solaris 11.2

### ISSUE THREAD

**Issue by stephanarts** [phusion/passenger#1701]
gem install passenger works; and then, if I try to compile the apache2 module, it fails.
```
Compiling and installing Apache 2 module...
cd /usr/ruby/2.1/lib/ruby/gems/2.1.0/gems/passenger-5.0.22
# env NOEXEC_DISABLE=1 /usr/ruby/2.1/bin/ruby /usr/ruby/2.1/bin/rake RELEASE=yes apache2:clean apache2
rm -rf buildout/cache
rm -rf buildout/common/libboost_oxt.a buildout/common/libboost_oxt
rm -f buildout/common/libpassenger_common/Logging.o buildout/common/libpassenger_common/Exceptions.o buildout/common/libpassenger_common/Utils/SystemTime.o buildout/common/libpassenger_common/Utils/StrIntUtils.o buildout/common/libpassenger_common/Utils/StrIntUtilsNoStrictAliasing.o buildout/common/libpassenger_common/Utils/IOUtils.o buildout/common/libpassenger_common/Utils.o
rm -f buildout/common/libpassenger_common/Utils/CachedFileStat.o buildout/common/libpassenger_common/Utils/LargeFiles.o buildout/common/libpassenger_common/WatchdogLauncher.o buildout/common/libpassenger_common/MemoryKit/mbuf.o buildout/common/libpassenger_common/MemoryKit/palloc.o buildout/common/libpassenger_common/ServerKit/http_parser.o buildout/common/libpassenger_common/ServerKit/Implementation.o buildout/common/libpassenger_common/Utils/Hasher.o buildout/common/libpassenger_common/AppTypes.o
rm -f buildout/common/libpassenger_common/jsoncpp.o
rm -f buildout/common/libpassenger_common/vendor-modified/modp_b64.o
rm -f buildout/common/libpassenger_common/UnionStationFilterSupport.o
rm -rf buildout/common/libpassenger_common
rm -rf buildout/support-binaries/
rm -rf buildout/apache2/module_libboost_oxt.a buildout/apache2/module_libboost_oxt
rm -f buildout/apache2/module_libpassenger_common/Logging.o buildout/apache2/module_libpassenger_common/Exceptions.o buildout/apache2/module_libpassenger_common/Utils/SystemTime.o buildout/apache2/module_libpassenger_common/Utils/StrIntUtils.o buildout/apache2/module_libpassenger_common/Utils/StrIntUtilsNoStrictAliasing.o buildout/apache2/module_libpassenger_common/Utils/IOUtils.o buildout/apache2/module_libpassenger_common/Utils.o
rm -f buildout/apache2/module_libpassenger_common/Utils/CachedFileStat.o buildout/apache2/module_libpassenger_common/Utils/LargeFiles.o buildout/apache2/module_libpassenger_common/WatchdogLauncher.o buildout/apache2/module_libpassenger_common/MemoryKit/mbuf.o buildout/apache2/module_libpassenger_common/MemoryKit/palloc.o buildout/apache2/module_libpassenger_common/ServerKit/http_parser.o buildout/apache2/module_libpassenger_common/ServerKit/Implementation.o buildout/apache2/module_libpassenger_common/Utils/Hasher.o buildout/apache2/module_libpassenger_common/AppTypes.o
rm -f buildout/apache2/module_libpassenger_common/jsoncpp.o
rm -f buildout/apache2/module_libpassenger_common/vendor-modified/modp_b64.o
rm -f buildout/apache2/module_libpassenger_common/UnionStationFilterSupport.o
rm -rf buildout/apache2/module_libpassenger_common
rm -rf buildout/apache2/mod_passenger.o buildout/apache2/Configuration.o buildout/apache2/Bucket.o buildout/apache2/Hooks.o buildout/apache2/mod_passenger.so
mkdir -p buildout/apache2/module_libpassenger_common
c++ -o buildout/apache2/module_libpassenger_common/Logging.o  -Isrc/cxx_supportlib -Isrc/cxx_supportlib/vendor-copy -Isrc/cxx_supportlib/vendor-modified -Isrc/cxx_supportlib/vendor-modified/libev -O -fPIC -DSOLARIS2=11 -D_POSIX_PTHREAD_SEMANTICS -D_REENTRANT -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/apr/1.5/include -I/usr/apr-util/1.5/include -I/usr/include/openldap -m64 -Ui386 -U__i386 -DSSL_EXPERIMENTAL -DSSL_ENGINE -I/usr/apache2/2.2/include -D_REENTRANT -I/usr/local/include -Wall -Wextra -Wno-unused-parameter -Wno-parentheses -Wpointer-arith -Wwrite-strings -Wno-long-long -Wno-missing-field-initializers -feliminate-unused-debug-symbols -feliminate-unused-debug-types -fvisibility=hidden -DVISIBILITY_ATTRIBUTE_SUPPORTED -ggdb -DHAS_ALLOCA_H -DHAS_SFENCE -DHAS_LFENCE -DPASSENGER_DEBUG -DBOOST_DISABLE_ASSERTS -std=gnu++11 -Wno-unused-local-typedefs -DHASH_NAMESPACE="__gnu_cxx" -DHASH_MAP_HEADER="<hash_map>" -DHASH_MAP_CLASS="hash_map" -DHASH_FUN_H="<hash_fun.h>" -pthreads -D_XOPEN_SOURCE=500 -D_XPG4_2 -D__EXTENSIONS__ -D__SOLARIS__ -D_FILE_OFFSET_BITS=64 -DBOOST_HAS_STDINT_H -c src/cxx_supportlib/Logging.cpp
<command-line>:0:0: warning: "_XOPEN_SOURCE" redefined [enabled by default]
src/cxx_supportlib/Logging.cpp:1:0: note: this is the location of the previous definition
 /*
 ^
In file included from /usr/include/wchar.h:8:0,
                 from /usr/gcc/4.8/include/c++/4.8.2/cwchar:44,
                 from /usr/gcc/4.8/include/c++/4.8.2/bits/postypes.h:40,
                 from /usr/gcc/4.8/include/c++/4.8.2/iosfwd:40,
                 from /usr/gcc/4.8/include/c++/4.8.2/iomanip:39,
                 from src/cxx_supportlib/Logging.cpp:26:
/usr/include/sys/feature_tests.h:354:2: error: #error "Compiler or options invalid for pre-UNIX 03 X/Open applications  and pre-2001 POSIX applications"
 #error "Compiler or options invalid for pre-UNIX 03 X/Open applications \
  ^
In file included from src/cxx_supportlib/vendor-modified/boost/system/system_error.hpp:14:0,
                 from src/cxx_supportlib/vendor-modified/boost/thread/exceptions.hpp:22,
                 from src/cxx_supportlib/vendor-modified/boost/thread/pthread/thread_data.hpp:10,
                 from src/cxx_supportlib/vendor-modified/boost/thread/thread_only.hpp:17,
                 from src/cxx_supportlib/vendor-modified/boost/thread/thread.hpp:12,
                 from src/cxx_supportlib/vendor-modified/boost/thread.hpp:13,
                 from src/cxx_supportlib/Logging.cpp:32:
src/cxx_supportlib/vendor-modified/boost/system/error_code.hpp:222:36: warning: ‘boost::system::posix_category’ defined but not used [-Wunused-variable]
     static const error_category &  posix_category = generic_category();
                                    ^
src/cxx_supportlib/vendor-modified/boost/system/error_code.hpp:223:36: warning: ‘boost::system::errno_ecat’ defined but not used [-Wunused-variable]
     static const error_category &  errno_ecat     = generic_category();
                                    ^
src/cxx_supportlib/vendor-modified/boost/system/error_code.hpp:224:36: warning: ‘boost::system::native_ecat’ defined but not used [-Wunused-variable]
     static const error_category &  native_ecat    = system_category();
                                    ^
rake aborted!
Command failed with status (1): [c++ -o buildout/apache2/module_libpassenger_common/Logging.o  -Isrc/cxx_supportlib -Isrc/cxx_supportlib/vendor-copy -Isrc/cxx_supportlib/vendor-modified -Isrc/cxx_supportlib/vendor-modified/libev -O -fPIC -DSOLARIS2=11 -D_POSIX_PTHREAD_SEMANTICS -D_REENTRANT -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/apr/1.5/include -I/usr/apr-util/1.5/include -I/usr/include/openldap -m64 -Ui386 -U__i386 -DSSL_EXPERIMENTAL -DSSL_ENGINE -I/usr/apache2/2.2/include -D_REENTRANT -I/usr/local/include -Wall -Wextra -Wno-unused-parameter -Wno-parentheses -Wpointer-arith -Wwrite-strings -Wno-long-long -Wno-missing-field-initializers -feliminate-unused-debug-symbols -feliminate-unused-debug-types -fvisibility=hidden -DVISIBILITY_ATTRIBUTE_SUPPORTED -ggdb -DHAS_ALLOCA_H -DHAS_SFENCE -DHAS_LFENCE -DPASSENGER_DEBUG -DBOOST_DISABLE_ASSERTS -std=gnu++11 -Wno-unused-local-typedefs -DHASH_NAMESPACE="__gnu_cxx" -DHASH_MAP_HEADER="<hash_map>" -DHASH_MAP_CLASS="hash_map" -DHASH_FUN_H="<hash_fun.h>" -pthreads -D_XOPEN_SOURCE=500 -D_XPG4_2 -D__EXTENSIONS__ -D__SOLARIS__ -D_FILE_OFFSET_BITS=64 -DBOOST_HAS_STDINT_H -c src/cxx_supportlib/Logging.cpp]

Tasks: TOP => apache2 => buildout/apache2/mod_passenger.so => buildout/apache2/module_libpassenger_common/Logging.o
(See full trace by running task with --trace)

--------------------------------------------

It looks like something went wrong

Please read our documentation for troubleshooting tips:

   https://www.phusionpassenger.com/library/install/apache/
   https://www.phusionpassenger.com/library/admin/apache/troubleshooting/

If that doesn't help, please use our support facilities. We'll do our best to help you.

   https://www.phusionpassenger.com/documentation_and_support
```

I've tried running passenger standalone aswell, but when compiling the agent it gives a simmilar error:

```
# passenger start --engine=builtin
 --> Downloading a Phusion Passenger agent binary for your platform
     Could not download https://oss-binaries.phusionpassenger.com/binaries/passenger/by_release/5.0.22/agent-x86-solaris.tar.gz: The requested URL returned error: 404 Not Found
     Trying next mirror...
     Could not download https://s3.amazonaws.com/phusion-passenger/binaries/passenger/by_release/5.0.22/agent-x86-solaris.tar.gz: The requested URL returned error: 403 Forbidden
     ------------------------------------------
     Sorry, no precompiled agent binary is available for your platform.

---------------------------------------

The Phusion Passenger agent binary could not be downloaded. Compiling it from source instead.

Checking for required software...

 * Checking for C compiler...
      Found: yes
      Location: /usr/gcc/4.8/bin/gcc
 * Checking for C++ compiler...
      Found: yes
      Location: /usr/ruby/2.1/bin/c++
 * Checking for Rake (associated with /usr/ruby/2.1/bin/ruby)...
      Found: yes
      Location: /usr/ruby/2.1/bin/ruby /usr/ruby/2.1/bin/rake
 * Checking for Curl development headers with SSL support...
      Found: yes
      curl-config location: /usr/bin/curl-config
      Header location: /usr/include/curl/curl.h
      Version: libcurl 7.40.0
      Usable: yes
      Supports SSL: yes
 * Checking for OpenSSL development headers...
      Found: yes
      Location: /usr/include/openssl/ssl.h
 * Checking for Zlib development headers...
      Found: yes
      Location: /usr/include/zlib.h


--------------------------------------------

Compile the agent with optimizations?

Compiling the agent with optimizations will make Phusion Passenger faster, but
it will take longer to compile and it requires at least 2 GB of memory.

Compile with optimizations? [y/n]: n

Compiling Phusion Passenger agent...
[**                                                ] -
*** ERROR: a Rake command failed. You can find the full log in /tmp/passenger-install-log-6b5a6f. Below, you can find the last few lines of the command's output.
------------- Begin command output snippet -------------
rake aborted!
Command failed with status (1): [c++ -o /tmp/passenger-install.19hdznv/support-binaries/AgentMain.o  -Isrc/agent -Isrc/cxx_supportlib -Isrc/cxx_supportlib/vendor-copy -Isrc/cxx_supportlib/vendor-modified -Isrc/cxx_supportlib/vendor-modified/libev -Isrc/cxx_supportlib/vendor-copy/libuv/include -D_REENTRANT -I/usr/local/include -Wall -Wextra -Wno-unused-parameter -Wno-parentheses -Wpointer-arith -Wwrite-strings -Wno-long-long -Wno-missing-field-initializers -feliminate-unused-debug-symbols -feliminate-unused-debug-types -fvisibility=hidden -DVISIBILITY_ATTRIBUTE_SUPPORTED -ggdb -DHAS_ALLOCA_H -DHAS_SFENCE -DHAS_LFENCE -DPASSENGER_DEBUG -DBOOST_DISABLE_ASSERTS -std=gnu++11 -Wno-unused-local-typedefs -DHASH_NAMESPACE="__gnu_cxx" -DHASH_MAP_HEADER="<hash_map>" -DHASH_MAP_CLASS="hash_map" -DHASH_FUN_H="<hash_fun.h>" -pthreads -D_XOPEN_SOURCE=500 -D_XPG4_2 -D__EXTENSIONS__ -D__SOLARIS__ -D_FILE_OFFSET_BITS=64 -DBOOST_HAS_STDINT_H -c src/agent/AgentMain.cpp]
/usr/ruby/2.1/lib/ruby/gems/2.1.0/gems/passenger-5.0.22/build/cplusplus_support.rb:52:in `run_compiler'
/usr/ruby/2.1/lib/ruby/gems/2.1.0/gems/passenger-5.0.22/build/cplusplus_support.rb:102:in `compile_cxx'
/usr/ruby/2.1/lib/ruby/gems/2.1.0/gems/passenger-5.0.22/build/cplusplus_support.rb:160:in `block in define_cxx_object_compilation_task'
/usr/ruby/2.1/lib/ruby/2.1.0/rake/task.rb:236:in `call'
/usr/ruby/2.1/lib/ruby/2.1.0/rake/task.rb:236:in `block in execute'
/usr/ruby/2.1/lib/ruby/2.1.0/rake/task.rb:231:in `each'
/usr/ruby/2.1/lib/ruby/2.1.0/rake/task.rb:231:in `execute'
/usr/ruby/2.1/lib/ruby/2.1.0/rake/task.rb:175:in `block in invoke_with_call_chain'
/usr/ruby/2.1/lib/ruby/2.1.0/monitor.rb:211:in `mon_synchronize'
/usr/ruby/2.1/lib/ruby/2.1.0/rake/task.rb:168:in `invoke_with_call_chain'
/usr/ruby/2.1/lib/ruby/2.1.0/rake/task.rb:197:in `block in invoke_prerequisites'
/usr/ruby/2.1/lib/ruby/2.1.0/rake/task.rb:195:in `each'
/usr/ruby/2.1/lib/ruby/2.1.0/rake/task.rb:195:in `invoke_prerequisites'
/usr/ruby/2.1/lib/ruby/2.1.0/rake/task.rb:174:in `block in invoke_with_call_chain'
/usr/ruby/2.1/lib/ruby/2.1.0/monitor.rb:211:in `mon_synchronize'
/usr/ruby/2.1/lib/ruby/2.1.0/rake/task.rb:168:in `invoke_with_call_chain'
/usr/ruby/2.1/lib/ruby/2.1.0/rake/task.rb:161:in `invoke'
/usr/ruby/2.1/lib/ruby/2.1.0/rake/application.rb:149:in `invoke_task'
/usr/ruby/2.1/lib/ruby/2.1.0/rake/application.rb:106:in `block (2 levels) in top_level'
/usr/ruby/2.1/lib/ruby/2.1.0/rake/application.rb:106:in `each'
/usr/ruby/2.1/lib/ruby/2.1.0/rake/application.rb:106:in `block in top_level'
/usr/ruby/2.1/lib/ruby/2.1.0/rake/application.rb:115:in `run_with_threads'
/usr/ruby/2.1/lib/ruby/2.1.0/rake/application.rb:100:in `top_level'
/usr/ruby/2.1/lib/ruby/2.1.0/rake/application.rb:78:in `block in run'
/usr/ruby/2.1/lib/ruby/2.1.0/rake/application.rb:165:in `standard_exception_handling'
/usr/ruby/2.1/lib/ruby/2.1.0/rake/application.rb:75:in `run'
/usr/ruby/2.1/bin/rake:33:in `<main>'
Tasks: TOP => /tmp/passenger-install.19hdznv/support-binaries/PassengerAgent => /tmp/passenger-install.19hdznv/support-binaries/AgentMain.o
------------- End command output snippet -------------
```

**Comment by eimann** [comment]
Same goes for passenger-install-nginx-module.
$ ruby -e 'puts RUBY_PLATFORM'         
x86_64-solaris2.11

SunOS v00020100274 5.11 joyent_20151104T185720Z i86pc i386 i86pc Solaris

**Comment by eimann** [comment]
Okay, so apparently in src/ruby_supportlib/phusion_passenger/platform_info/cxx_portability.rb this isn't working as it should:

        if os_name =~ /solaris2\.11/
          # skip the _XOPEN_SOURCE and _XPG4_2 definitions in later versions of Solaris / OpenIndiana
          flags << '-D__EXTENSIONS__ -D__SOLARIS__ -D_FILE_OFFSET_BITS=64'

It wouldn't append the flag at all.

HAVE_INOTIFY_INIT and HAVE_SYS_INOTIFY_H for libev should be unset too. After those changes it compiles with nginx 1.9.9 source just fine.

**Comment by OnixGH** [comment]
Thanks guys. This looks like a regression in 5.0.22 caused by the fix for #1643. The `os_name` is stripped so the versionmatching no longer works, resulting in the flag getting appended. I'll start work on a correction.

**Comment by OnixGH** [comment]
@stephanarts @eimann can you see if [branch GH-1701](https://github.com/phusion/passenger/tree/GH-1701) solves your compilation problem?

**Comment by OnixGH** [comment]
The fix was verified for #1694, so I think it's fixed here too.
